terror/terror.go

```diff
@@ -144,6 +128,15 @@ func (ec ErrClass) NotEqualClass(err error) bool {
 // New creates an *Error with an error code and an error message.
 // Usually used to create base *Error.
 func (ec ErrClass) New(code ErrCode, message string) *Error {
+	if ErrClassToMySQLCodes == nil {
+		ErrClassToMySQLCodes = make(map[ErrClass]map[ErrCode]struct{})
```

Comment:
Any chance this function is called simultaneously from two different goroutines?

Reply:
no this, function only can be called in `init()` or `global variable initializer`,  I will add some comments for this :D

Reply:
But there is no need for this lazy initialization @lysu